utils/oscap-remediate-offline-service

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (C) 2021 Red Hat Inc., Durham, North Carolina.
+#
+# Evgenii Kolesnikov <ekolesni@redhat.com>
+#
+# The offline system remediation (systemd's /system-update) service wrapper
+
+log() {
+    echo "${1}" | logger -t oscap-remediate-offline -p ${2:-info}
+}
+
+config=$(readlink -f /system-update)
+
+if [[ "$(basename ${config})" != "oscap-remediate-offline.conf.sh" ]]; then
+    log "The /system-update symlink was not created by oscap-remediate-offline, ignoring"
+    systemd-notify --ready --status="Skipped"
+    exit 0
+fi
+
+rm -f /system-update
+
+log "Found config: ${config}"
+
+source "$config"
+
+log "Data stream: ${OSCAP_REMEDIATE_DS}"
+log "Profile ID: ${OSCAP_REMEDIATE_PROFILE_ID}"
+
+#TODO: Check
+
+profile_title_line=$(oscap info --profile "${OSCAP_REMEDIATE_PROFILE_ID}" "${OSCAP_REMEDIATE_DS}" | grep Title)
+
+#TODO: Check
+
+profile_title=${profile_title_line#*Title: }
+
+log "Profile name: ${profile_title}"
+
+oscap xccdf validate --skip-schematron "${OSCAP_REMEDIATE_DS}"
+
+#TODO: Check
+
+log_dir="/var/lib/ssg/$(date +%Y.%m.%d-%H:%M:%S)"
+
+mkdir -p "${log_dir}" && cd "${log_dir}"
+
+#TODO: Check?
+
+command="oscap xccdf eval --profile ${OSCAP_REMEDIATE_PROFILE_ID}\
+    --results xccdf-results.xml\
+    --results-arf arf-results.xml\
+    --report report.html\
+    --progress-full\
+    --remediate ${OSCAP_REMEDIATE_DS}"
+
+while IFS= read -r line; do
+	if [[ "${line}" =~ "---evaluation" ]]; then
+		message="OpenSCAP is checking the system for compliance using"$'\n'"${profile_title}"$'\n\n'
+		log "Evaluating..."
+		continue
+	fi
+	if [[ "${line}" =~ "---remediation" ]]; then
+		message="OpenSCAP is remediating the system to be compliant using the"$'\n'"${profile_title}"$'\n\n'
+		log "Remediating..."
+		continue
+	fi
+	# TODO: Format is "rule_id|Rule Title|result"
+	plymouth display-message --text="${message}${line#*	}"
+	log "${line#*	}"
+done < <(${command})
+
```

Comment:
In my `journalctl` output I can see multiple messages like this:

```
Jan 06 14:52:18 fedora oscap-remediate-offline-service[13975]: plymouth: ply-boot-client.c:465: ply_boot_client_queue_request: Assertion `request_argument == NULL || strlen (request_argument) <= UCHAR_MAX' failed.
Jan 06 14:52:18 fedora oscap-remediate-offline-service[554]: /usr/libexec/oscap-remediate-offline-service: line 72: 13975 Aborted                 (core dumped) plymouth display-message --text="${message}${line#*        }"
```

Reply:
Yeah, so we'd better cut down the text on the boot screen. The whole message with the profile name and nice wording and current rule title hitting the message length limit :(

Reply:
Fixed